crates/holochain/src/conductor/conductor.rs

```diff
@@ -648,11 +647,34 @@ where
 
         let p2p_kv = AgentKv::new(environ.clone().into())?;
         let env = environ.guard();
-        let reader = env.reader()?;
 
-        Ok(p2p_kv
-            .as_store_ref()
-            .get(&reader, &(&*kitsune_space, &*kitsune_agent).into())?)
+        let res = {
+            let reader = env.reader()?;
```

Comment:
One thing to keep in mind is that if new agent info is added between this read and the below `.with_commit()` that deletes the info then the new info will be lost. Basically the read will be stale.
You probably want to get the agent info and delete it in the same transaction.

Reply:
good call, but unfortunate... now all peer requests require a write lock : ( - but we can proceed with this for now.